test/e2e/pod_schedule_error_test.go

```diff
@@ -0,0 +1,117 @@
+		Image:   "helloworld",
+	}
+
+	defer test.TearDown(clients, names)
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+
+	t.Logf("Creating a new Service %s", names.Image)
+	resources := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU: resource.MustParse("50000"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("350Mi"),
+		},
+	}
+	var (
+		svc *v1alpha1.Service
+		err error
+	)
+	if svc, err = test.CreateLatestService(t, clients, names, &test.Options{ContainerResources: resources}); err != nil {
+		t.Fatalf("Failed to create Service %s: %v", names.Service, err)
+	}
+
+	names.Config = serviceresourcenames.Configuration(svc)
+
+	err = test.WaitForServiceState(clients.ServingClient, names.Service, func(r *v1alpha1.Service) (bool, error) {
+		cond := r.Status.GetCondition(v1alpha1.ConfigurationConditionReady)
+		if cond != nil && !cond.IsUnknown() {
+			if strings.Contains(cond.Message, errorMsg) && cond.IsFalse() {
+				return true, nil
+			}
+			t.Logf("Reason: %s ; Message: %s ; Status: %s", cond.Reason, cond.Message, cond.Status)
+			return true, fmt.Errorf("The service %s was not marked with expected error condition (Reason=\"%s\", Message=\"%s\", Status=\"%s\"), but with (Reason=\"%s\", Message=\"%s\", Status=\"%s\")",
+				names.Config, errorReason, errorMsg, "False", cond.Reason, cond.Message, cond.Status)
+		}
+		return false, nil
+	}, "ContainerUnscheduleable")
+
+	if err != nil {
+		t.Fatalf("Failed to validate service state: %s", err)
+	}
+
+	revisionName, err := revisionFromConfiguration(clients, names.Config)
+	if err != nil {
+		t.Fatalf("Failed to get revision from configuration %s: %v", names.Config, err)
+	}
+
+	t.Log("When the containers are not scheduled, the revision should have error status.")
+	err = test.WaitForRevisionState(clients.ServingClient, revisionName, func(r *v1alpha1.Revision) (bool, error) {
+		cond := r.Status.GetCondition(v1alpha1.RevisionConditionReady)
+		if cond != nil {
+			if cond.Reason == revisionReason && strings.Contains(cond.Message, errorMsg) {
+				return true, nil
+			}
+			return true, fmt.Errorf("The revision %s was not marked with expected error condition (Reason=%q, Message=%q), but with (Reason=%q, Message=%q)",
+				revisionName, revisionReason, errorMsg, cond.Reason, cond.Message)
+		}
+		return false, nil
+	}, "RevisionFailed")
+
+	if err != nil {
+		t.Fatalf("Failed to validate revision state: %s", err)
+	}
+}
+
+// Get revision name from configuration.
+func revisionFromConfiguration(clients *test.Clients, configName string) (string, error) {
+	config, err := clients.ServingClient.Configs.Get(configName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	if config.Status.LatestCreatedRevisionName != "" {
+		return config.Status.LatestCreatedRevisionName, nil
+	}
+	return "", fmt.Errorf("No valid revision name found in configuration %s", configName)
```

Comment:
errors start with lowercase in Go.
```suggestion
	return "", fmt.Errorf("no valid revision name found in configuration %s", configName)
```